BurnOutSharp/ProtectionType/Tages.cs

```diff
@@ -20,110 +20,190 @@ public string CheckContents(string file, byte[] fileContent, bool includeDebug,
-                        0x64, 0x2D, 0x74, 0x61, 0x67, 0x65, 0x73, 0x2D,
-                        0x72, 0x75, 0x6E, 0x74, 0x69, 0x6D, 0x65, 0x2E,
-                        0x65, 0x78, 0x65
-                    }, Utilities.GetFileVersion, "TAGES"),
-
-                    // (char)0xE8 + u + (char)0x00 + (char)0x00 + (char)0x00 + (char)0xE8
-                    new ContentMatchSet(new byte?[] { 0xE8, 0x75, 0x00, 0x00, 0x00, 0xE8 }, GetVersion, "TAGES"),
-                };
-
-                return MatchUtil.GetFirstMatch(file, fileContent, contentMatchSets, includeDebug);
-            }
+                    0x70, 0x72, 0x6F, 0x74, 0x65, 0x63, 0x74, 0x65,
+                    0x64, 0x2D, 0x74, 0x61, 0x67, 0x65, 0x73, 0x2D,
+                    0x72, 0x75, 0x6E, 0x74, 0x69, 0x6D, 0x65, 0x2E,
+                    0x65, 0x78, 0x65
+                }, Utilities.GetFileVersion, "TAGES [DEBUG]"),
 
-            return null;
+                // This check seems to currently be broken, as files that appear to have this string aren't being detected.
+                // (char)0xE8 + u + (char)0x00 + (char)0x00 + (char)0x00 + (char)0xE8
+                new ContentMatchSet(new byte?[] { 0xE8, 0x75, 0x00, 0x00, 0x00, 0xE8 }, GetVersion, "TAGES [DEBUG]"),
+            };
+            
+            return MatchUtil.GetFirstMatch(file, fileContent, contentMatchSets, includeDebug);
         }
-
+        
         /// <inheritdoc/>
         public ConcurrentQueue<string> CheckDirectoryPath(string path, IEnumerable<string> files)
         {
-            var protections = new ConcurrentQueue<string>();
+            var matchers = new List<PathMatchSet>
+            {
+                new PathMatchSet(new List<string>
+                {
+                    /*
+                    So far, only known to exist in early versions of "Moto Racer 3".
+
+                    Expected information about these checks:
+                    ./sys/Devx.sys (d37f70489207014d7d0fbaa43b081a93e8030498)
+                    ./sys/VtPr.sys (a0acbc2f8e321e4f30c913c095e28af444058249)
+                    ./Wave.aif (SHA-1 is variable, file size is 81,920 bytes)
+                    ./Wave.alf (f82339d797be6da92f5d9dadeae9025385159057)
+                    ./Wave.apt (0351d0f3d4166362a1a9d838c9390a3d92945a44)
+                    ./Wave.axt (SHA-1 is variable, file size is 61,440 bytes)
+                    */
+                    Path.Combine("Sys", "Devx.sys").Replace("\\", "/"),
+                    Path.Combine("Sys", "VtPr.sys").Replace("\\", "/"),
+                    Path.Combine("Wave.aif").Replace("\\", "/"),
+                    Path.Combine("Wave.alf").Replace("\\", "/"),
+                    Path.Combine("Wave.apt").Replace("\\", "/"),
+                    Path.Combine("Wave.axt").Replace("\\", "/"),
+                }, "TAGES"),
+
+                new PathMatchSet(new List<string>
+                {
+                    /*
```

Comment:
Ditto

Reply:
Additionally, when you group these checks up, it means that _all checks_ in one of these sets need to pass in order to be detected. I don't think that's what you want for the last 2 groups.